hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/CreatePrecompileSuite.java

```diff
@@ -115,16 +144,570 @@ List<HapiSpec> positiveSpecs() {
                 tokenCreateWithKeyWithMultipleKeyValuesReverts(),
                 tokenCreateWithFixedFeeWithMultiplePaymentsReverts(),
                 createTokenWithEmptyTokenStruct(),
                 createTokenWithInvalidExpiry(),
                 createTokenWithInvalidTreasury(),
                 createTokenWithInsufficientValueSent(),
+                createTokenWithFixedFeeThenTransferAndAssessFee(),
                 delegateCallTokenCreateFails());
     }
 
+    // TEST-001
+    @HapiTest
+    final HapiSpec fungibleTokenCreateHappyPath() {
+        final var tokenCreateContractAsKeyDelegate = "tokenCreateContractAsKeyDelegate";
+        final var createTokenNum = new AtomicLong();
+        final AtomicReference<byte[]> ed2551Key = new AtomicReference<>();
+        return defaultHapiSpec("fungibleTokenCreateHappyPath")
+                .given(
+                        newKeyNamed(ECDSA_KEY).shape(SECP256K1),
+                        newKeyNamed(CONTRACT_ADMIN_KEY),
+                        cryptoCreate(ACCOUNT_TO_ASSOCIATE),
+                        uploadInitCode(TOKEN_CREATE_CONTRACT),
+                        cryptoCreate(ACCOUNT).balance(ONE_MILLION_HBARS),
+                        contractCreate(TOKEN_CREATE_CONTRACT)
+                                .autoRenewAccountId(ACCOUNT)
+                                .adminKey(CONTRACT_ADMIN_KEY)
+                                .gas(GAS_TO_OFFER),
+                        newKeyNamed(THRESHOLD_KEY)
+                                .shape(THRESHOLD_KEY_SHAPE.signedWith(sigs(ED25519_ON, TOKEN_CREATE_CONTRACT)))
+                                .exposingKeyTo(k -> ed2551Key.set(k.getThresholdKey()
+                                        .getKeys()
+                                        .getKeys(0)
+                                        .getEd25519()
+                                        .toByteArray())),
+                        cryptoUpdate(ACCOUNT).key(THRESHOLD_KEY),
+                        cryptoUpdate(ACCOUNT_TO_ASSOCIATE).key(THRESHOLD_KEY))
+                .when(withOpContext((spec, opLog) -> {
+                    spec.registry()
+                            .saveKey(
+                                    ED25519KEY,
+                                    spec.registry()
+                                            .getKey(THRESHOLD_KEY)
+                                            .getThresholdKey()
+                                            .getKeys()
+                                            .getKeys(0));
+                    spec.registry()
+                            .saveContractId(
+                                    "tokenCreateContracId", spec.registry().getContractId(TOKEN_CREATE_CONTRACT));
+                    allRunFor(
+                            spec,
+                            contractCall(
+                                            TOKEN_CREATE_CONTRACT,
+                                            "createTokenWithKeysAndExpiry",
```

Comment:
Maybe a constant for this string?  I think it is used below as well.

Reply:
I extracted it as a constant, thanks